vector-stores/spring-ai-pgvector-store/src/main/java/org/springframework/ai/vectorstore/PgVectorStore.java

```diff
@@ -535,12 +558,35 @@ public Builder withInitializeSchema(boolean initializeSchema) {
 			return this;
 		}
 
+		public Builder withObservationRegistry(ObservationRegistry observationRegistry) {
+			this.observationRegistry = observationRegistry;
+			return this;
+		}
+
+		public Builder withSearchObservationConvention(
+				VectorStoreObservationConvention customSearchObservationConvention) {
+			this.searchObservationConvention = customSearchObservationConvention;
+			return this;
+		}
+
 		public PgVectorStore build() {
 			return new PgVectorStore(schemaName, vectorTableName, vectorTableValidationsEnabled, jdbcTemplate,
 					embeddingModel, dimensions, distanceType, removeExistingVectorStoreTable, indexType,
-					initializeSchema);
+					initializeSchema, observationRegistry, searchObservationConvention);
 		}
 
 	}
 
+	@Override
+	public VectorStoreObservationContext.Builder createObservationContextBuilder(String operationName) {
+
+		return VectorStoreObservationContext.builder("pg_vector", operationName)
+			.withDimensions(this.embeddingDimensions())
+			.withCollectionName(this.vectorTableName)
+			.withNamespace(this.schemaName)
+			.withSimilarityMetric(this.getDistanceType().name())
```

Comment:
The OTel Conventions have defined a list of well known values for the similarity metric. I would suggest defining a `VectorStoreSimilarityMetric` enum in the same package as `AiOperationType` and use those values if possible. That will allow LLM observability backend to recognise the known values. Here, we would need to map between the DB specific name and the convention name.